elements/lisk-chain/src/data_access/data_access.ts

```diff
@@ -61,8 +66,29 @@ export class DataAccess {
 		return this._blocksCache.add(blockHeader);
 	}
 
-	public removeBlockHeader(id: string): BlockHeader[] {
-		return this._blocksCache.remove(id);
+	public async removeBlockHeader(id: string): Promise<BlockHeader[]> {
+		const cachedItems = this._blocksCache.remove(id);
+
+		if (this._blocksCache.needsRefill) {
+			// Get the height limits to fetch
+			// The method getBlocksByHeightBetween uses gte & lte so we need to adjust values
+			const upperHeightToFetch = this._blocksCache.items[0]?.height - 1 || 0;
+
+			const lowerHeightToFetch = Math.max(
+				upperHeightToFetch -
+					(this._blocksCache.maxCachedItems - this._blocksCache.minCachedItems),
+				1,
+			);
+
+			const blockHeaders = await this.getBlocksByHeightBetween(
```

Comment:
Before calling this function, it's probably better to check if `upperHeight - lowerHeight > 0` and in that case we don't need to fetch DB

Reply:
Done.